src/vec.rs

```diff
@@ -858,6 +888,17 @@ impl<T, const N: usize> Drop for Vec<T, N> {
     }
 }
 
+impl<T, const N: usize> From<[T; N]> for Vec<T, N> {
```

Comment:
it feels a bit strange that the array length becomes the capacity. Intuitively I'd say it should be possible to convert a `[u8; 10]` into a `Vec<u8; 20>` with 10 filled elements.

This would fail if `CAP < N`, and there's no way to add `where CAP >= N` in const generics, so tihs'd have to be TryFrom which perhaps is not that ergonomic?

Reply:
There is a way to generate a compile error in that case. This PR does this: https://github.com/rust-embedded/heapless/pull/352

Reply:
I thought doing it this way would allow me to "move" the array into the Vec without copying, but it turns out that copying will happen either way, so might as well allow N < CAP.